### DISCUSSION TITLE
how to unCompress gzip request with httpclient?

### DISCUSSION THREAD

**Discussion by baloyou** [esp8266/Arduino#8909] [Q&A]
I need to make a GET request via "HTTPClient", but the data is GZIP compressed, how can I decompress it?

This code would have been normal.But the API was changed to gzip form, and the received content became garbled.
```
if (https.begin(client, "https://devapi.qweather.com/v7/weather/now?location=101070102&key=mykey")) {  // HTTPS
      Serial.print("[HTTPS] GET...\n");
      int httpCode = https.GET();

      if (httpCode > 0) {
        Serial.printf("[HTTPS] GET... code: %d\n", httpCode);

        if (httpCode == HTTP_CODE_OK || httpCode == HTTP_CODE_MOVED_PERMANENTLY) {
          String payload = https.getString();
          Serial.println(payload);
        }
      } else {
        Serial.printf("[HTTPS] GET... failed, error: %s\n", https.errorToString(httpCode).c_str());
      }
      
      https.end();
```

**Comment by earlephilhower** [comment]
You can't, for most practical purposes.  The smallest GZip library out there, uzlib, needs a 32K contiguous dictionary buffer plus a few K of other buffers and structures.  You're unlikely to have that much free, contiguous RAM at runtime.
